src/models/service/transactions/mod.rs

```diff
@@ -91,8 +91,16 @@ pub struct SettingsChange {
 
 #[derive(Serialize, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
 pub enum SettingsInfo {
```

Comment:
When do we need structs inside the enum and when not? (I always forget, but there were specific cases when this was required)

Reply:
I am unsure too now. Maybe this is something we could homogenise in a refactor after the release. Generally if you use tuple enums you don't get to fiddle with the fields casings. Also, enum variants themselves cannot be used as types. So if you want to return a specific type with a method, like we do when we map from backend transaction to service transaction, a nested struct is better, there we can differentiate types and return `Transfer`, `SettingsChange` or `Custom` and not always `ServiceTransaction` . You cannot have a variant of an enum as a return type.